### ISSUE TITLE
Mq routing keys

### ISSUE THREAD

**Issue by allanchau** [idearium/developers#25]
The current MQ routing keys have a different format to the queues which can lead to confusion.

I would like to propose the following updated format:

`{project}.{container}.{file-name}`

This would simplify the naming structure, for example:

``` javascript
// Publisher.
const exchange = 'updated';
const routingKey = 'project.membermanagementcontainer.user-created';

// Consumer.
const exchange = 'updated';
const queue = 'project.userportalcontainer.user-created';
const routingKey = 'project.membermanagementcontainer.user-created';
```

To compare the current format is as follows (note the dot in the routingKey):
``` javascript
// Publisher.
const exchange = 'updated';
const routingKey = 'project.membermanagementcontainer.user.created';

// Consumer.
const exchange = 'updated';
const queue = 'project.userportalcontainer.user-created';
const routingKey = 'project.membermanagementcontainer.user.created';
```

**Comment by allanchau** [comment]
@smebberson @patelswapnil What do you guys think?

**Comment by patelswapnil** [comment]
@allanchau, I agree that {file-name} is more clear then {file.name}. Recently I have named the routing key as {project}.{container}.{file-name} in MMS, and it is more readable.

**Comment by smebberson** [comment]
@patelswapnil, @allanchau: The thing that I like about the current routing keys is that you can very effectively use pattern topics to listen to specific types of messages, have a read here https://www.rabbitmq.com/tutorials/tutorial-five-python.html.

Using the above naming structure won't allow this. For example, how would I listen only to all user events in one consumer? Or how would I listen to only edit events, across multiple models?
